modules/core/src/main/java/org/apache/ignite/internal/logger/IgniteThrottledLoggerImpl.java

```diff
@@ -0,0 +1,345 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.logger;
+
+import static org.apache.ignite.internal.util.IgniteUtils.capacity;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import java.lang.System.Logger;
+import java.lang.System.Logger.Level;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Supplier;
+import org.apache.ignite.internal.lang.IgniteStringFormatter;
+import org.apache.ignite.internal.util.FastTimestamps;
+import org.jetbrains.annotations.Nullable;
+
+class IgniteThrottledLoggerImpl implements IgniteThrottledLogger {
+    /** Logger delegate. */
+    private final System.Logger delegate;
+
+    /** Log messages. */
+    private final Map<LogThrottleKey, Long> messagesMap = Caffeine.newBuilder()
+            .initialCapacity(capacity(128))
+            .maximumSize(128)
```

Comment:
Why maximum size and not eviction by last access time?

Reply:
In general, why not.